documentation/specs/static-graph.md

```diff
@@ -227,34 +140,107 @@ We'll represent the project reference protocols as items in MSBuild. The target
 </ItemGroup>
 ```
 
-## Building a single project
-The project graph is required to build the projects in the correct order. An individual project will be built using `/p:BuildProjectReferences=false` to avoid recursive builds inside a project.
+### Underspecified graphs
+The intention is that the project graph and the target lists for each node be exactly correct, however MSBuild is quite flexible and particular projects or project types may not adequately decribe these for the project graph.
+
+If a project calls into another project which either isn't represented in the graph or with a target list which isn't represented by the graph, it will fall back to classical MSBuild behavior and execute that target on the project reference just-in-time. This has the consequence of still requiring all project state be kept in memory in case any arbitrary project wants to execute targets on any other arbitrary project.
+
+To enable further optimizations (and strictness), graph builds can run [isolated](#isolated-builds) which enforces that the graph be entirely accurate.
+
+### Public API
+This is a proposal for what the public API for ProjectGraph may look like:
+
+    namespace Microsoft.Build.Graph
+    {
+        public class ProjectGraph
+        {
+            // Creates a graph starting at the given project file.
+            public ProjectGraph(string projectFile) { }
+            public ProjectGraph(string entryProjectFile, IDictionary<string, string> globalProperties) { }
+
+            // Creates a graph starting at the given project files.
+            public ProjectGraph(IEnumerable<string> projectFiles) { }
+            public ProjectGraph(IEnumerable<string> entryProjectFiles, IDictionary<string, string> globalProperties) { }
```

Comment:
Does this mean that the provided global properties are repeated for each entry point file?